ts/components/services/ContactPreferencesToggles/index.tsx

```diff
@@ -1,91 +1,174 @@
 import I18n from "../../../i18n";
 import ItemSeparatorComponent from "../../ItemSeparatorComponent";
 import { NotificationChannelEnum } from "../../../../definitions/backend/NotificationChannel";
+import { Dispatch } from "../../../store/actions/types";
+import {
+  servicePreferenceSelector,
+  ServicePreferenceState
+} from "../../../store/reducers/entities/services/servicePreference";
+import { ServiceId } from "../../../../definitions/backend/ServiceId";
+import {
+  loadServicePreference,
+  upsertServicePreference
+} from "../../../store/actions/services/servicePreference";
+import { serviceIDCurrentSelector } from "../../../store/reducers/entities/services/currentService";
+import {
+  isServicePreferenceResponseSuccess,
+  ServicePreference
+} from "../../../types/services/ServicePreferenceResponse";
+import { isStrictSome } from "../../../utils/pot";
+import { showToast } from "../../../utils/showToast";
 import PreferenceToggleRow from "./PreferenceToggleRow";
 
 type Item = "email" | "push" | "inbox";
 
 type Props = {
   channels?: ReadonlyArray<NotificationChannelEnum>;
-} & ReturnType<typeof mapStateToProps>;
+} & ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
+
+const hasChannel = (
+  channel: NotificationChannelEnum,
+  channels?: ReadonlyArray<NotificationChannelEnum>
+) =>
+  fromNullable(channels)
+    .map(anc => anc.indexOf(channel) !== -1)
+    .getOrElse(true);
+
+const getValueOrFalse = (
+  potServicePreference: ServicePreferenceState,
+  key: Item
+): boolean => {
+  if (
+    isStrictSome(potServicePreference) &&
+    isServicePreferenceResponseSuccess(potServicePreference.value)
+  ) {
+    return potServicePreference.value.value[key];
+  }
+  return false;
+};
 
 const ContactPreferencesToggle: React.FC<Props> = (props: Props) => {
-  // functions from actions
-  const onValueChange: (item: Item, value: boolean) => void = (_item, _value) =>
-    undefined;
-
-  /*
-   *  TODO State should be removed when data will be available from store
-   */
-  const [inboxSwitched, setInboxSwitched] = useState(true);
-  const [pushSwitched, setPushSwitched] = useState(true);
-  const [emailSwitched, setEmailSwitched] = useState(true);
-
-  const onInboxValueChange = (value: boolean) => {
-    onValueChange("inbox", value);
-    setInboxSwitched(value);
+  const { isLoading, isError } = props;
+  const [isFirstRender, setIsFirstRender] = useState(true);
+
+  const loadPreferences = () => {
+    if (props.currentService !== null) {
+      props.loadServicePreference(props.currentService.serviceID);
+    }
   };
 
-  const hasChannel = (channel: NotificationChannelEnum) =>
-    fromNullable(props.channels)
-      .map(anc => anc.indexOf(channel) !== -1)
-      .getOrElse(true);
+  useEffect(loadPreferences, []);
 
-  const onPushValueChange = (value: boolean) => {
-    onValueChange("push", value);
-    setPushSwitched(value);
-  };
+  useEffect(() => {
+    if (!isFirstRender) {
```

Comment:
I'm not sure I'm getting this part: we don't want to show the error if this is the first render. 
Can you please confirm that this is correct and perhaps elaborate a bit as to why we need it?

Reply:
Without this check would happen this scenario:
- user sets tries to set a toggle to false
- API answers an error -> toast appears
- user navigate back to list
- user navigate back to a service detail
- a toast appear for the previous error state
- preferences load again
- API answers 200

the toast should not appear if the state is previously evaluated with an error